src/components/Dashboard/DashboardItem/AppItem.js

```diff
@@ -23,33 +28,69 @@ class AppItem extends ItemBase {
 
     static create(props) {
         const item = new AppItem(props);
-        const { t } = useTranslation("dashboard");
+        const { showErrorAnnouncer } = props;
+        const { t } = useTranslation(["dashboard", "apps"]);
+
+        // Extract app details. Note: dashboard-aggregator only queries the DE database.
+        const app = props.content;
+
+        // State variables.
+        const [isFavorite, setIsFavorite] = useState(app.is_favorite);
+
+        // Functions to build keys and links.
+        const baseId = `${constants.KIND_APPS}-${app.system_id}-${app.id}`;
+        const buildKey = (keyType) => `${baseId}-${keyType}`;
+        const buildRef = (refType) =>
```

Comment:
Minor nitpick, but I might name this `buildHRef` or `buildPath` or something, only because React already uses the `ref` term for DOM element refs, so I was confused for an instant (see https://reactjs.org/docs/refs-and-the-dom.html for example).